test/com/arpnetworking/metrics/portal/TestBeanFactory.java

```diff
@@ -0,0 +1,134 @@
+                .setName(TEST_NAME)
+                .setOperator(TEST_OPERATOR)
+                .setPeriod(Period.seconds(TEST_PERIOD_IN_SECONDS))
+                .setStatistic(TEST_STATISTIC)
+                .setValue(TEST_QUANTITY);
+    }
+
+    public static Alert createAlert() {
+        return createAlertBuilder().build();
+    }
+
+    public static models.ebean.Alert createEbeanAlert() {
+        final models.ebean.Alert ebeanAlert = new models.ebean.Alert();
+        ebeanAlert.setUuid(UUID.randomUUID());
+        ebeanAlert.setNagiosExtension(createEbeanNagiosExtension());
+        ebeanAlert.setName(TEST_NAME);
+        ebeanAlert.setOperator(TEST_OPERATOR);
+        ebeanAlert.setPeriod(TEST_PERIOD_IN_SECONDS);
+        ebeanAlert.setStatistic(TEST_STATISTIC);
+        ebeanAlert.setQuantityValue(TEST_QUANTITY_VALUE);
+        ebeanAlert.setQuantityUnit(TEST_QUANTITY_UNIT);
+        ebeanAlert.setCluster(TEST_CLUSTER);
+        ebeanAlert.setMetric(TEST_METRIC);
+        ebeanAlert.setContext(TEST_CONTEXT);
+        ebeanAlert.setService(TEST_SERVICE);
+        return ebeanAlert;
+    }
+
+    public static models.internal.NagiosExtension createNagiosExtension() {
+        return new models.internal.NagiosExtension.Builder()
+                .setSeverity(TEST_NAGIOS_SEVERITY)
+                .setNotify(TEST_NAGIOS_NOTIFY)
+                .setMaxCheckAttempts(TEST_MAX_CHECK_ATTEMPTS)
+                .setFreshnessThresholdInSeconds(TEST_FRESHNESS_THRESHOLD)
+                .build();
+    }
+
+    public static NagiosExtension createEbeanNagiosExtension() {
+        final NagiosExtension nagiosExtension = new NagiosExtension();
+        nagiosExtension.setSeverity(TEST_NAGIOS_SEVERITY);
+        nagiosExtension.setNotify(TEST_NAGIOS_NOTIFY);
+        nagiosExtension.setMaxCheckAttempts(TEST_MAX_CHECK_ATTEMPTS);
+        nagiosExtension.setFreshnessThreshold(TEST_FRESHNESS_THRESHOLD);
+        return nagiosExtension;
+    }
+
+    public static Expression createEbeanExpression() {
+        final models.ebean.Expression ebeanExpression = new models.ebean.Expression();
+        ebeanExpression.setUuid(UUID.randomUUID());
+        ebeanExpression.setCluster(TEST_CLUSTER);
+        ebeanExpression.setMetric(TEST_METRIC);
+        ebeanExpression.setScript(TEST_SCRIPT);
+        ebeanExpression.setService(TEST_SERVICE);
+        return ebeanExpression;
+    }
+
+    public static DefaultExpression.Builder createExpressionBuilder() {
+        return new DefaultExpression.Builder()
+                .setId(UUID.randomUUID())
+                .setCluster(TEST_CLUSTER)
+                .setMetric(TEST_METRIC)
+                .setScript(TEST_SCRIPT)
+                .setService(TEST_SERVICE);
+    }
+
+    public static models.internal.Expression createExpression() {
+        return createExpressionBuilder().build();
+    }
+
+    private static final String TEST_CLUSTER = "test-cluster";
```

Comment:
All this data should be randomly generated. The point is the default values should _never_ be relied on by a test. That is, if a test wants to assert something about the service name for example it should set it to a specific value. The bean factory here is not intended to create well-known beans because: 1) having more tests eventually diverge the requirements for the bean and thus reduce its effectiveness; 2) it hides the test's preconditions/dependencies by hiding the specific data values in use. Happy to discuss this further.